internal/brokers/examplebroker/dbus.go

```diff
@@ -47,21 +49,38 @@ func StartBus() (err error) {
 			},
 		},
 	}), dbusObjectPath, introspect.IntrospectData.Name); err != nil {
-		_ = conn.Close()
 		return err
 	}
 
 	reply, err := conn.RequestName(dbusInterface, dbus.NameFlagDoNotQueue)
 	if err != nil {
-		_ = conn.Close()
 		return err
 	}
 	if reply != dbus.RequestNameReplyPrimaryOwner {
-		_ = conn.Close()
 		return fmt.Errorf("D-Bus name already taken")
 	}
 
-	select {}
+	// Write the broker configuration file.
+	err = os.MkdirAll(configPath, 0750)
+	if err != nil {
+		return err
+	}
+	err = os.WriteFile(filepath.Join(configPath, "examplebroker.conf"),
+		[]byte(`
+name = ExampleBroker
+brand_icon = /usr/share/backgrounds/warty-final-ubuntu.png
+
+[dbus]
+name = com.ubuntu.authd.ExampleBroker
+object = /com/ubuntu/authd/ExampleBroker
+interface = com.ubuntu.authd.ExampleBroker`),
+		0600)
+	if err != nil {
+		os.RemoveAll(configPath)
+	}
+
+	<-ctx.Done()
+	return os.RemoveAll("/etc/authd")
```

Comment:
I thought about removing only the config file as well, but I decided to remove everything for now since this main function actually runs on "our" machine and I thought it wouldn't be great to have the leftover path.

But I guess the logic to remove the path could be transferred to the main func and then the broker logic gets more concise. Wdyt?

Reply:
I'd love to, but the path is only overridable through a config file so, if the path is not under `/etc/authd/broker.d`, the manager can not recognize the broker :smiling_face_with_tear: